### PR TITLE
Pipeline: Support pipeline table scan in UT

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #6518 

Problem Summary:
Currently only support table scan in unit test.
Will run table scan in ft when exchange is supported in pipeline

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Operators/UnorderedSourceOp.cpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                {
+                    continue;
+                }
+            }
+            else
+            {
+                return OperatorStatus::FINISHED;
+            }
+        }
+    }
```
Comment (SeaRise) [inline dbms/src/Operators/UnorderedSourceOp.cpp:59]:
```suggestion
    auto await_status = awaitImpl();
    if (await_status == OperatorStatus::HAS_OUTPUT)
    {
        // handle t_block
    }
    return await_status;
```

Comment (ywqzzy) [reply dbms/src/Operators/UnorderedSourceOp.cpp:59]: Good point!!

File: dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp [hunk 2]
```diff
@@ -1062,6 +1063,79 @@ BlockInputStreams DeltaMergeStore::read(const Context & db_context,
...
+        expected_block_size,
+        getReadMode(db_context, is_fast_scan, keep_order),
+        std::move(tasks),
+        after_segment_read,
+        log_tracing_id,
+        enable_read_thread,
+        final_num_stream);
+
+    SourceOps res;
+    RUNTIME_CHECK(enable_read_thread); // TODO: support keep order
```
Comment (SeaRise) [inline dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp:1122]: And add `!executor.keep_order()` in `Pipeline::isSupport()`

File: dbms/src/Operators/UnorderedSourceOp.cpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    if (await_status == OperatorStatus::HAS_OUTPUT)
+    {
+        std::swap(block, t_block.value());
+        t_block.reset();
+        if (action.transform(block))
+        {
+            return OperatorStatus::HAS_OUTPUT;
+        }
+        else
+            return OperatorStatus::FINISHED;
```
Comment (SeaRise) [inline dbms/src/Operators/UnorderedSourceOp.cpp:31]:
```suggestion
        return action.transform(block) ? OperatorStatus::HAS_OUTPUT : OperatorStatus::FINISHED;
```

File: dbms/src/Operators/UnorderedSourceOp.cpp [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    while (true)
+    {
+        Block res;
+        if (!task_pool->tryPopBlock(res))
+            return OperatorStatus::WAITING;
+        if (res)
+        {
+            if (res.rows() == 0)
+                continue;
+            t_block = std::move(res);
```
Comment (SeaRise) [inline dbms/src/Operators/UnorderedSourceOp.cpp:49]:
```suggestion
            if unlikely (res.rows() == 0)
                continue;
            t_block.emplace(std::move(res));
```

File: dbms/src/Operators/UnorderedSourceOp.h [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        ref_no = task_pool->increaseUnorderedInputStreamRefCount();
+        LOG_DEBUG(log, "Created, pool_id={} ref_no={}", task_pool->poolId(), ref_no);
+        addReadTaskPoolToScheduler();
+    }
+
+    String getName() const override
+    {
+        return "UnorderedSourceOp";
+    }
+
```
Comment (SeaRise) [inline dbms/src/Operators/UnorderedSourceOp.h:56]:
```suggestion

protected:
```

File: dbms/src/Operators/UnorderedSourceOp.h [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    OperatorStatus readImpl(Block & block) override;
+
+    OperatorStatus awaitImpl() override;
+
```
Comment (SeaRise) [inline dbms/src/Operators/UnorderedSourceOp.h:60]:
```suggestion

private:
```

**Review by SeaRise — APPROVED** [review]
Others LGTM

File: dbms/src/Debug/MockStorage.cpp [hunk 7]
```diff
@@ -130,7 +130,7 @@ Int64 MockStorage::addTableDataForDeltaMerge(Context & context, const String & n
 
         // write data to DeltaMergeStorage
         ASTPtr insertptr(new ASTInsertQuery());
-        BlockOutputStreamPtr output = storage->write(insertptr, context.getSettingsRef());
+        BlockOutputStreamPtr output = storage->write(insertptr, context.getGlobalContext().getSettingsRef());
```
Comment (SeaRise) [inline dbms/src/Debug/MockStorage.cpp:133]: why change it?

File: dbms/src/Operators/UnorderedSourceOp.h [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+#include <Common/Logger.h>
+#include <DataStreams/SegmentReadTransformAction.h>
+#include <Operators/Operator.h>
+#include <Storages/DeltaMerge/ReadThread/SegmentReadTaskScheduler.h>
+#include <Storages/DeltaMerge/SegmentReadTaskPool.h>
+
+namespace DB
+{
+class UnorderedSourceOp : public SourceOp
```
Comment (SeaRise) [inline dbms/src/Operators/UnorderedSourceOp.h:25]: How about adding some comments here?

File: dbms/src/Debug/MockStorage.cpp (hunk 7 above)
Comment (ywqzzy) [reply dbms/src/Debug/MockStorage.cpp]: > why change it?

Updated

**Pull request merged**